### ISSUE TITLE
False Positive for RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE in try-with-resources

### ISSUE THREAD

**Issue by victornoel** [spotbugs/spotbugs#1694]
Getting this error with spotbugs 4.4.1 but not with 4.2.2.
```
[ERROR] Medium: Nullcheck of reader at line 18 of value previously dereferenced in net.pricefx.remotejob.io.core.FP.safelyReadObject(InputStream) [default.FP, default.FP] At FP.java:[line 18]Redundant null check at FP.java:[line 19] RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE
```

```java
import java.io.InputStream;

import javax.json.Json;
import javax.json.JsonObject;

public class FP {

    private final JsonObject json;

    public FP(InputStream json) {
        this(safelyReadObject(json));
    }

    public FP(JsonObject json) {
        this.json = json;
    }

    private static JsonObject safelyReadObject(InputStream json) {
        try (var reader = Json.createReader(json)) {
            return reader.readObject();
        }
    }
}
```

**Comment by marcelmore** [comment]
If you are on JDK11 or later, this might be a duplicate of https://github.com/spotbugs/spotbugs/issues/1338.

**Comment by victornoel** [comment]
@marcelmore thank you, you must be correct yes! Since it's a different pattern, I will keep this open, watch the other issue and close this one if fixing the other fixes this one!

**Comment by marcelmore** [comment]
@victornoel Sorry, I was wrong. https://github.com/spotbugs/spotbugs/issues/1338 triggers RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE but your code triggers RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE.

**Comment by marcelmore** [comment]
I also got multiple warnings of RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE.
Updating from 4.3.0 to 4.4.1.
Example:

````java
package dummy;

import org.apache.poi.ss.usermodel.Workbook;
import org.apache.poi.ss.usermodel.WorkbookFactory;

import java.io.File;
import java.io.IOException;

public class Dummy {
    public void tryReadItemFilter() throws IOException {
        try (Workbook wb = WorkbookFactory.create(new File(""), null, true)) {
            wb.getNumberOfSheets();
        } // triggerd here: Redundant null check at Dummy.java:[line 13]
    }
}

````

**Comment by victornoel** [comment]
@marcelmore maybe they are related in terms of implementation though :) let's see what happens!

**Comment by victornoel** [comment]
Still present in 4.5.0

**Comment by dmivankov** [comment]
@Paharov you mentioned in https://github.com/spotbugs/spotbugs/pull/1932#issuecomment-1125789427
that #1932 doesn't resolve the issue for you
Could you try to run following on latest spotbugs checkout?
```console
gradlew assemble

java -cp "./spotbugs/build/libs/spotbugs.jar:./spotbugs-annotations/build/libs/*:./spotbugs/.libs/*"  edu.umd.cs.findbugs.LaunchAppropriateUI -textui -longBugCodes -onlyAnalyze "ghIssues.issue600.*" spotbugsTestCases/build/libs/spotbugsTestCases-4.7.1-SNAPSHOT.jar
```
and check what inspection it produces, it'd scan in particular https://github.com/dmivankov/spotbugs/blob/master/spotbugsTestCases/src/java/ghIssues/issue600/TryWithResourcesSimple.java class that no longer triggers false-positive in master but triggers in 4.7.0.
Then if that command works you could try
```console
java -cp "./spotbugs/build/libs/spotbugs.jar:./spotbugs-annotations/build/libs/*:./spotbugs/.libs/*" edu.umd.cs.findbugs.LaunchAppropriateUI -textui -longBugCodes YOUR_JAR_HERE
```
and/or adding minimized example of code that is still false-positive into TryWithResourcesSimple.java and checking what happens

**Comment by mofleury** [comment]
Hi,

FYI, this bug seems to be fixed.

I tested the code sample against versions 4.4.1 and 4.73 ; I have false positive bugs on 4.4.1 but none on 4.7.3.

Hope that helps!

**Comment by JuditKnoll** [comment]
Thanks, @mofleury! Then I'll close this issue.
